pkg/hive/job/job_test.go

```diff
@@ -455,15 +459,20 @@ func TestTimer_DoubleTrigger(t *testing.T) {
 		l.Append(g)
 	})
 
+	// Trigger a few times before we start consuming events, these should coalesce
+	trigger.Trigger()
+	trigger.Trigger()
+	trigger.Trigger()
+
 	if err := h.Start(context.Background()); err != nil {
 		t.Fatal(err)
 	}
 
-	trigger.Trigger()
-	trigger.Trigger()
-	trigger.Trigger()
 	<-ran
 
+	// Give the timer some time for bogus additional runs to come in after the first one
+	time.Sleep(time.Second)
```

Comment:
Is this strictly necessary? Sleeping in tests is a big hammer.

Reply:
Ideally I would be able to check if the timer thread is currently blocked or not. I will remove this and the 100ms timer and then hope that when the behavior under test breaks that the callback is called more then once between `<-ran` and `Stop`.

Testing stuff like this is just hard without time